x-pack/legacy/plugins/fleet/public/pages/policy_details/components/edit_policy.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState } from 'react';
+import {
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiButton,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { Policy } from '../../../../scripts/mock_spec/types';
+import { PolicyForm } from '../../../components/policy_form';
+import { useLibs } from '../../../hooks/use_libs';
+
+interface RouterProps {
+  policy: Policy;
+  onClose: () => void;
+}
+
+export const EditPolicyFlyout: React.FC<RouterProps> = ({ policy: originalPolicy, onClose }) => {
+  const libs = useLibs();
+
+  const [policy, setPolicy] = useState<Partial<Policy>>({
+    name: originalPolicy.name,
+    description: originalPolicy.description,
+  });
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const updatePolicy = (updatedFields: Partial<Policy>) => {
+    setPolicy({
+      ...policy,
+      ...updatedFields,
+    });
+  };
+
+  const renderHeader = () => (
```

Comment:
nitpick: you probably do not need a function here and you can just do `const header = (<EuiFlyoutHeader> ... </EuiFlyoutHeader>)`